client/crypto-sdk/lib/address.cpp

```diff
@@ -14,18 +15,34 @@
 using namespace std;
 using namespace Orbs;
 
+const uint8_t Address::MAIN_NETWORK_ID = 0x14; // "M" in BASE58.
+const uint8_t Address::TEST_NETWORK_ID = 0x1A; // "T" in BASE58;
+
 const uint8_t Address::VERSION = 0;
 const uint32_t Address::PUBLIC_KEY_SIZE = 32;
-const uint32_t Address::VERSION_FIELD_SIZE = 1;
+const uint32_t Address::NETWORK_ID_SIZE = 1;
+const uint32_t Address::VERSION_SIZE = 1;
+const uint32_t Address::VIRTUAL_CHAIN_ID_SIZE = 3;
+const uint8_t Address::VIRTUAL_CHAIN_ID_MSB = 0x08;
 const uint32_t Address::ACCOUNT_ID_SIZE = 20;
 const uint32_t Address::CHECKSUM_SIZE = 4;
-const uint32_t Address::ADDRESS_LENGTH = 34;
+const uint32_t Address::ADDRESS_LENGTH = 39;
 
-Address::Address(const vector<uint8_t> &publicKey) : publicKey_(publicKey) {
+Address::Address(const vector<uint8_t> &publicKey, const vector<uint8_t> &virtualChainId, uint8_t networkId) :
+    publicKey_(publicKey), virtualChainId_(virtualChainId), networkId_(networkId) {
     Init();
 }
 
-Address::Address(const string &publicKey) : publicKey_(Utils::Hex2Vec(publicKey)) {
+Address::Address(const string &publicKey, const string &virtualChainId, const std::string &networkId) :
+    publicKey_(Utils::Hex2Vec(publicKey)), virtualChainId_(Utils::Hex2Vec(virtualChainId))  {
+
+    vector<uint8_t> decodedNetworkId = Base58::Decode(networkId);
```

Comment:
I find this decoding an overkill

Reply:
Not fixed yet

Reply:
I've written below - this will be changed in the future, because: 1) Context switch; 2) It was done according to the spec; 3) There more pending changes and this one will take 2-4 hours to do, since all the test addressed will need to be regenerated.